app/models/wpcc/contributions_calendar.rb

```diff
@@ -1,60 +1,28 @@
 module Wpcc
   class ContributionsCalendar
-    attr_reader :eligible_salary,
-                :salary_frequency,
-                :employee_percent,
-                :employer_percent
-
-    PERIODS_FILE = Wpcc::Engine.root.join('config', 'contribution_periods.yml')
-    PERIODS = HashWithIndifferentAccess.new(YAML.load_file(PERIODS_FILE))
-
-    def initialize(eligible_salary:,
-                   employee_percent:,
-                   employer_percent:,
-                   salary_frequency:)
-
-      @eligible_salary = eligible_salary
-      @employee_percent = employee_percent
-      @employer_percent = employer_percent
-      @salary_frequency = salary_frequency
-    end
+    include ActiveModel::Model
+    attr_accessor :eligible_salary,
+                  :salary_frequency,
+                  :employee_percent,
+                  :employer_percent
 
     def schedule
-      periods_above_user_contributions.map do |period, percents|
-        employee_percentage = percents[:employee_percent] || employee_percent
-        employer_percentage = percents[:employer_percent] || employer_percent
-
+      legal_periods_percents.map do |period|
```

Comment:
I'm not sure if it is good naming. Not sure if it's just me, but I can't immediately process what `legal_periods_percents` is..
The argument in the `.map` block suggests it's a list of periods.. to me `legal_periods_percents` tells me it's a list of percents from some "legal periods"... which isn't correct, is it?